InterGalacticCurrencyConvertor/src/main/java/DataProcessor.java

```diff
@@ -1,80 +1,73 @@
 package main.java;
 
-import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 
-import main.java.query.Query;
-import main.java.statement.Statement;
+import main.java.query.processor.InterGalacticToDecimalQueryProcessor;
+import main.java.query.processor.MaterialCostQueryProcessor;
+import main.java.query.processor.QueryProcessor;
+import main.java.statement.processor.InterGalacticToRomanMapStatementProcessor;
+import main.java.statement.processor.MaterialCostStatementProcessor;
+import main.java.statement.processor.StatementProcessor;
 
 /**
  * Processes user input based on its types: Statement or Query.
  */
 public class DataProcessor {
-	private static ArrayList<Query> mQueryObjList = new ArrayList<>();
-	private static ArrayList<Statement> mStatementObjList = new ArrayList<>();
+  private static DataProcessor mInstance = new DataProcessor();
+	private static List<QueryProcessor> mQueryProcessorList;
```

Comment:
Can these two lists be merged into a single list? (Also see comment below about using a single interface for both query and statement).

Reply:
done